File: market/integration-test/java/ru/yandex/market/logistics/werewolf/AbstractTest.java
package ru.yandex.market.logistics.werewolf;

import org.assertj.core.api.SoftAssertions;
import org.assertj.core.api.junit.jupiter.InjectSoftAssertions;
import org.assertj.core.api.junit.jupiter.SoftAssertionsExtension;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.extension.ExtendWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.boot.test.mock.mockito.MockitoTestExecutionListener;
import org.springframework.boot.test.mock.mockito.ResetMocksTestExecutionListener;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.TestExecutionListeners;
import org.springframework.test.context.support.DependencyInjectionTestExecutionListener;
import org.springframework.test.web.servlet.MockMvc;

import ru.yandex.market.logistics.werewolf.config.IntegrationTestConfiguration;
import ru.yandex.market.request.trace.RequestContext;
import ru.yandex.market.request.trace.RequestContextHolder;

@ActiveProfiles("integration-test")
@SpringBootTest(
    classes = IntegrationTestConfiguration.class,
    webEnvironment = SpringBootTest.WebEnvironment.MOCK,
    properties = "spring.config.name=integration-test"
)
@AutoConfigureMockMvc
@TestExecutionListeners({
    MockitoTestExecutionListener.class,
    ResetMocksTestExecutionListener.class,
    DependencyInjectionTestExecutionListener.class,
})
@ComponentScan({
    "ru.yandex.market.logistics.werewolf.config",
    "ru.yandex.market.logistics.werewolf.controller",
    "ru.yandex.market.logistics.werewolf.util",
    "ru.yandex.market.logistics.werewolf.facade",
    "ru.yandex.market.logistics.werewolf.converter",
})
@ExtendWith(SoftAssertionsExtension.class)
public abstract class AbstractTest {
    protected static final String REQUEST_ID = "1000000000000/abcdabcdabcdabcdabcdabcdabcdabcd";

    @Autowired
    protected MockMvc mockMvc;

    @InjectSoftAssertions
    protected SoftAssertions softly;

    @BeforeEach
    void setup() {
        RequestContextHolder.setContext(new RequestContext(REQUEST_ID));
    }

    @AfterEach
    void tearDown() {
        RequestContextHolder.clearContext();
    }
}

File: market/test/java/ru/yandex/market/dsm/core/test/AbstractTest.kt
package ru.yandex.market.dsm.core.test

import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.jdbc.AutoConfigureTestDatabase
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.ActiveProfiles
import org.springframework.test.context.TestPropertySource
import org.springframework.test.context.junit.jupiter.SpringExtension
import ru.yandex.market.dsm.config.DbConfiguration
import ru.yandex.market.dsm.config.DbQueueConfiguration
import ru.yandex.market.dsm.config.DsmConstants
import ru.yandex.market.dsm.config.MockConfiguration
import ru.yandex.market.dsm.config.TestConfiguration
import ru.yandex.market.dsm.test.CleanupAfterEachExtension
import ru.yandex.market.javaframework.main.config.SpringApplicationConfig
import ru.yandex.market.starter.postgres.config.EmbeddedPostgresConfiguration
import java.time.Clock

@ActiveProfiles(DsmConstants.ENV.FUNCTIONAL_TEST_PROFILE)
@ExtendWith(value = [
    SpringExtension::class,
    CleanupAfterEachExtension::class
])
@SpringBootTest(
    webEnvironment = SpringBootTest.WebEnvironment.NONE,
    classes = [
        SpringApplicationConfig::class,
        EmbeddedPostgresConfiguration::class,
        DbQueueConfiguration::class,
        DbConfiguration::class,
        TestConfiguration::class,
        MockConfiguration::class
    ]
)
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
@TestPropertySource(
    properties = [
        "spring.autoconfigure.exclude=ru.yandex.market.starter.quartz.config.MjQuartzAutoConfiguration",
        "quartzStubsEnabled=true",
        "tpl.dbQueue.runQueueLoop=false",
        "spring.jpa.properties.org.hibernate.envers.audit_table_suffix=_history",
        "spring.jpa.properties.org.hibernate.envers.revision_field_name=revision_id",
        "spring.jpa.properties.org.hibernate.envers.revision_type_field_name=revision_type",
        "spring.jpa.properties.org.hibernate.envers.store_data_at_delete=true",
        "spring.jpa.properties.org.hibernate.envers.audit_strategy=org.hibernate.envers.strategy.DefaultAuditStrategy",
        "courier.lobroker.producer.test-email-patterns=tpl-auto-user-.*@yandex.ru",
        "external.balance.url=http://test",
        "external.balance.tvm.applicationId=1"
    ]
)
abstract class AbstractTest {
    @Autowired
    private lateinit var clock: Clock

    @AfterEach
    fun resetClock() {
        ClockUtil.reset(clock)
    }
}
